CMakeLists.txt

```diff
@@ -661,7 +661,11 @@ if(NOT MSVC)
   # Details at http://eigen.tuxfamily.org/bz/show_bug.cgi?id=1459
   string(APPEND CMAKE_CXX_FLAGS " -Wall")
   string(APPEND CMAKE_CXX_FLAGS " -Wextra")
-  string(APPEND CMAKE_CXX_FLAGS " -Werror=return-type")
+  if($ENV{CMAKE_BUILD_WITH_WERROR})
+    string(APPEND CMAKE_CXX_FLAGS " -Werror")
+  else()
+    string(APPEND CMAKE_CXX_FLAGS " -Werror=return-type")
+  endif()
```

Comment:
Hmm, isn't it already overriding the logic in:
https://github.com/pytorch/pytorch/blob/e8c268746b297efa988e03abc61ff22203bf3980/CMakeLists.txt#L715-L721

Also, it's a bit unconventional to change build settings based on environment variable, adding it as CMAKE option would be a better idea